realm/index.d.ts

```diff
@@ -0,0 +1,356 @@
+        /**
+         * @returns Object|void
+         */
+        pop(): Object | void;
+
+        /**
+         * @param  {any} object
+         * @returns number
+         */
+        push(object: any): number;
+
+        /**
+         * @returns Object|void
+         */
+        shift(): Object | void;
+
+        /**
+         * @param  {number} index
+         * @param  {number} count?
+         * @param  {any} object?
+         * @returns Object
+         */
+        splice(index: number, count?: number, object?: any): Object[];
+
+        /**
+         * @param  {any} object
+         * @returns number
+         */
+        unshift(object: any): number;
+    }
+
+    /**
+    * Results
+    * @see { @link https://realm.io/docs/react-native/latest/api/Realm.Results.html }
+    */
+    export interface Results extends Collection {}
+}
+
+declare class Realm {
+    static defaultPath: string;
+
+    readonly path: string;
+
+    readonly readOnly: boolean;
+
+    readonly schema: Realm.ObjectSchema[];
+
+    /**
+     * @param  {string} path?
+     * @param  {any} encryptionKey?
+     * @returns number
+     */
+    schemaVersion(path?: string, encryptionKey?: any): number;
+
+    /**
+     * @param  {Realm.Configuration} config?
+     */
+    constructor(config?: Realm.Configuration);
+
+    /**
+     * @returns void
+     */
+    close(): void;
+
+    /**
+     * @param  {string|Realm.ObjectType} type
+     * @param  {Realm.ObjectPropsType} properties
+     * @param  {boolean} update?
+     * @returns Realm.Object|T|any
+     */
+    create<T>(type: string | Realm.ObjectType, properties: Realm.ObjectPropsType, update?: boolean): Realm.Object | T | any;
+
+    /**
+     * @param  {Realm.Object|Realm.Object[]|Realm.List|Realm.Results|any} object
+     * @returns void
+     */
+    delete(object: Realm.Object | Realm.Object[] | Realm.List | Realm.Results | any): void;
+
+    /**
+     * @returns void
+     */
+    deleteAll(): void;
+
+    /**
+     * @param  {string|Realm.ObjectType} type
+     * @param  {number|string} key
+     * @returns Realm.Object|void
+     */
+    objectForPrimaryKey(type: string | Realm.ObjectType, key: number | string): Realm.Object | void;
+
+    /**
+     * @param  {string|Realm.ObjectType} type
+     * @returns Realm.Results
+     */
+    objects(type: string | Realm.ObjectType): Realm.Results;
+
+    /**
+     * @param  {string} name
+     * @param  {()=>void} callback
+     * @returns void
+     */
+    addListener(name: string, callback: () => void): void;
+
+    /**
+     * @param  {string} name
+     * @param  {()=>void} callback
+     * @returns void
+     */
+    removeListener(name: string, callback: () => void): void;
+
+    /**
+     * @param  {string[]} name?
+     * @returns void
+     */
+    removeAllListeners(name?: string[]): void;
+
+    /**
+     * @param  {()=>void} callback
+     * @returns void
+     */
+    write(callback: () => void): void;
+}
+
+declare module 'realm' {
```

Comment:
could you strip `declare module` ?
https://github.com/Microsoft/types-publisher#fields-in-datadefinitionsjson
> UMD: Only top-level import and export declarations, as well as a UMD declaration. Preferred

https://github.com/Microsoft/TypeScript/issues/7125